nifi-toolkit/nifi-toolkit-tls/src/main/java/org/apache/nifi/toolkit/tls/diagnosis/TlsToolkitGetDiagnosisStandalone.java

```diff
@@ -0,0 +1,664 @@
+    }
+
+    public static void printAndThrowParsingException(String errorMessage, ExitCode exitCode) throws CommandLineParseException {
+        printUsage(errorMessage);
+        throw new CommandLineParseException(errorMessage, exitCode);
+    }
+
+    private static void displaySummaryReport() {
+        int correct = 0, wrong = 0, needsAttention = 0;
+        System.out.println("\n***********STANDALONE DIAGNOSIS SUMMARY***********\n");
+        for (Map.Entry<String, Tuple<String, Output>> each : outputSummary.entrySet()) {
+            String output = each.getValue().getValue().toString();
+            String type = StringUtils.rightPad(each.getKey(), 12);
+            System.out.println(type + " ==>   " + each.getValue().getKey());
+            switch (output) {
+                case "WRONG":
+                    wrong++;
+                    break;
+                case "CORRECT":
+                    correct++;
+                    break;
+                case "NEEDS_ATTENTION":
+                    needsAttention++;
+                    break;
+            }
+        }
+        System.out.println("\nCORRECT checks:         " + correct + "/7");
+        System.out.println("WRONG checks:           " + wrong + "/7");
+        System.out.println("NEEDS ATTENTION checks: " + needsAttention + "/7");
+        System.out.println("**************************************************\n");
+    }
+
+
+    public static void main(String[] args) {
+        TlsToolkitGetDiagnosisStandalone standalone = new TlsToolkitGetDiagnosisStandalone();
+
+        // TODO: If -v was added, change the logging config value
+
+        //Parse
+        try {
+            standalone.parseCommandLine(args);
+            standalone.niFiProperties = standalone.loadNiFiProperties();
+        } catch (CommandLineParseException e) {
+            System.exit(e.getExitCode().ordinal());
+        } catch (IOException e) {
+            printUsage(e.getLocalizedMessage());
+            System.exit(-1);
+        }
+
+        //Get keystore and truststore path
+        standalone.keystorePath = standalone.niFiProperties.getProperty("nifi.security.keystore");
+        standalone.truststorePath = standalone.niFiProperties.getProperty("nifi.security.truststore");
+        char[] keystorePassword = standalone.niFiProperties.getProperty("nifi.security.keystorePasswd").toCharArray();
+        standalone.keystoreType = standalone.niFiProperties.getProperty("nifi.security.keystoreType");
+        standalone.truststoreType = standalone.niFiProperties.getProperty("nifi.security.truststoreType");
+        char[] truststorePassword = standalone.niFiProperties.getProperty("nifi.security.truststorePasswd").toCharArray();
+
+        //Verify keystore and truststore are located at the correct file path
+        if ((doesFileExist(standalone.keystorePath, standalone.niFiPropertiesPath, ".jks")
+                && doesFileExist(standalone.truststorePath, standalone.niFiPropertiesPath, ".jks"))) {
```

Comment:
I think for toolkit we are only using .jks . I referred this link: https://nifi.apache.org/docs/nifi-docs/html/walkthroughs.html#securing-nifi-with-provided-certificates 
= "PKCS12 keystores are usable by NiFi, but JKS format is handled more robustly and causes fewer edge cases. JKS keystores cannot be formed directly from PEM files, so the PKCS12 keystore serves as an intermediate form"
Since this is for common deployment scenario, for now I have used only .jks keystores. This can be something to add further. Adding it as TODO.

Reply:
That sounds good.  The NiFi documentation should be updated as Java 9 changed the default preferred key store type to PKCS12 as described in [JEP 229](http://openjdk.java.net/jeps/229).